pgpy/packet/fields.py

```diff
@@ -345,6 +359,18 @@ def from_signer(self, sig):
         self.s = MPI(seq[1])
 
 
+class EdDSASignature(DSASignature):
+    def from_signer(self, sig):
+        lsig = len(sig)
+        assert lsig % 2 == 0
+        self.r = MPI(self.bytes_to_int(sig[:lsig // 2]))
+        self.s = MPI(self.bytes_to_int(sig[lsig // 2:]))
```

Comment:
do the work on this once instead of twice
```python
split = lsig // 2
self.r = MPI(self.bytes_to_int(sig[:split]))
self.s = MPI(self.bytes_to_int(sig[split:]))
```